cloudfront-function/main.tf

```diff
@@ -0,0 +1,55 @@
+/**
+ * Config
+ *
+ **/
+
+locals {
+  cloudfront_function_configs = {
+    usage   = var.cloudfront_function_config.usage
+    runtime = var.cloudfront_function_config.runtime
+    code = (
+      var.cloudfront_function_config.usage == "basic_auth" ?
+      file("${path.module}/function.tftpl") :
+      var.cloudfront_function_config.code
+    )
+    basic_auth = {
+      username = try(var.cloudfront_function_config.basic_auth.username, null) != null ? var.cloudfront_function_config.basic_auth.username : var.domain
```

Comment:
using a domain as a username is not intuitive. an empty string would be more understandable, but the best approach is to make the username required